content/blog/three-new-tidymodels-packages/index.Rmd

```diff
@@ -0,0 +1,192 @@
+
+With filtro, we use the pre-made score objects for these measures along with the `fit()` method: 
+
+```{r}
+#| label: fit-cor
+yield_rank_res <-
+  score_cor_spearman |>
+  fit(yield ~ ., data = yield_train)
+
+# The object contains the statistics:
+yield_rank_res@results |> 
+  arrange(desc(abs(score)))
+``` 
+
+It is just as simple for the random forest model
+
+```{r}
+#| label: fit-orf
+yield_rf_res <-
+  score_imp_rf_oblique |>
+  fit(yield ~ ., data = yield_train)
+
+yield_rf_res@results |> 
+  arrange(desc(abs(score)))
+```
+
+We should probably combine the scores and do a joint ranking. To combine the two sets of statistics: 
+
+```{r}
+#| label: bind
+
+class_score_list <-
+  list(
+    yield_rank_res,
+    yield_rf_res
+  ) |>
+  bind_scores()
+
+class_score_list
+```
+
+We can accomplish a joint ranking via desirability functions. Here, we set goals for each score (i.e., maximize, minimize, etc.). The algorithm rescales their values and uses a geometric mean for an overall ranking. The desirability2 package has some nice tools for this. Here's how we do it: 
+
+```{r}
+#| label: desirability
+library(desirability2)
+class_score_list |>
+  show_best_desirability_prop(
+    maximize(cor_spearman, low = 0.25, high = 1),
+    maximize(imp_rf_oblique, scale = 2)
+  ) |> 
+  arrange(desc(.d_overall)) |> 
+  select(-starts_with(".d_max_"))
+```
+
+Using the `scale = 2` options puts more weight on the random forest results. 
+
+Now that we've looked at filtro, next up is the important package (yes, this is what we named it). 
+
+## important
+
+The important package does two things. First, it provides yet another tool for calculating random forest-like permutation importance scores. We highly value other packages that perform these same calculations (such as [DALEX](https://modeloriented.github.io/DALEX/) and [vip](https://github.com/koalaverse/vip/)). Our rationale for creating another package for this is that we've developed interfaces for censored regression, including dynamic metrics such as Brier scores or ROC curves that evaluate models at a specific time point. These dynamic methods aren't available in other packages, and the peculiarities of these metrics make them difficult to incorporate into existing frameworks. 
+
+Other niceties about importance scores are that any metric from the yardstick methods can be used, and we have optimized parallel processing for the underlying computations. For the latter feature, we support the future and mirai packages for parallel processing. 
+
+important also has three recipe steps for supervised feature selection (similar to what Steven Pawley did with his [colino package](https://stevenpawley.github.io/colino/)). The steps are:
```

Comment:
I view this as the key element because that's what makes it easy to click these tools into your analysis workflow.